engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/selectModulesScreen/SelectModulesScreen.java

```diff
@@ -359,46 +362,89 @@ public String get() {
-                        }
+			for (CheckboxAssociation checkboxAssociation : CheckboxAssociation.values()) {
+				String checkboxName = checkboxAssociation.getCheckboxName();
+				StandardModuleExtension standardModuleExtension = checkboxAssociation.getStandardModuleExtension();
+
+				UICheckbox checkBox = find(checkboxName, UICheckbox.class);
+				if (null != checkBox) {
+					checkBox.setChecked(selectModulesConfig.isStandardModuleExtensionSelected(standardModuleExtension));
+					checkBox.subscribe(e -> {
+						selectModulesConfig.toggleStandardModuleExtensionSelected(standardModuleExtension);
+						checkBox.setChecked(selectModulesConfig.isStandardModuleExtensionSelected(standardModuleExtension));
+						filterModules();
+					});
+				} else {
+					logger.error("Unable to find checkbox named " + checkboxName + " in " + ASSET_URI.toString());
+					selectModulesConfig.unselectStandardModuleExtension(standardModuleExtension);
+				}
+			}
+
+			UICheckbox localOnlyCheckbox = find("localOnlyCheckbox", UICheckbox.class);
+			localOnlyCheckbox.setChecked(selectModulesConfig.isLocalOnlyChecked());
+			localOnlyCheckbox.subscribe(e -> {
+				selectModulesConfig.setIsLocalOnlyChecked(!selectModulesConfig.isLocalOnlyChecked());
+				localOnlyCheckbox.setChecked(selectModulesConfig.isLocalOnlyChecked());
+				filterModules();
+			});
+        }
+        WidgetUtil.trySubscribe(this, "close", button -> triggerBackAnimation());
+    }
 
-                        @Override
-                        public void set(Boolean value) {
-                            selectModulesConfig.setIsChecked(value);
-                            filterText(moduleSearch.getText());
-                            prepareModuleList(value);
-                        }
-                    }
-            );
+     private void filterModules() {
+        sortedModules.clear();
+        sortedModules.addAll(allSortedModules);
+        if(selectModulesConfig.isAnyStandardModuleExtensionSelected()) {
+            advancedModuleFilter();
+        }
+        if (selectModulesConfig.isLocalOnlyChecked()) {
+            localModuleFilter();
         }
+        filterText();
+    }
 
-        WidgetUtil.trySubscribe(this, "close", button -> triggerBackAnimation());
+    private void localModuleFilter() {
+        Iterator<ModuleSelectionInfo> iter = sortedModules.iterator();
+        while (iter.hasNext()) {
+            if (!iter.next().isPresent()) {
+                iter.remove();
+            }
+        }
     }
 
-    private void prepareModuleList(boolean checked) {
-        if (selectModulesConfig.isChecked()) {
-            Iterator<ModuleSelectionInfo> iter = sortedModules.iterator();
-            while (iter.hasNext()) {
-                if (!iter.next().isPresent()) {
-                    iter.remove();
-                }
+    private void advancedModuleFilter() {
+        Iterator<ModuleSelectionInfo> iter = sortedModules.iterator();
```

Comment:
Not thrilled about using an iterator and iter.remove, but that's what the code was doing before.